app/controllers/application_controller.rb

```diff
@@ -1,3 +1,18 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
+  
+  around_action :switch_locale
+
+  def switch_locale(&action)
+    logger.debug "* Accept-Language: #{request.env['HTTP_ACCEPT_LANGUAGE']}"
+    locale = extract_locale_from_accept_language_header
+    logger.debug "* Locale set to '#{locale}'"
+    I18n.with_locale(locale, &action)
+  end
+
+  private
+    def extract_locale_from_accept_language_header
+      request.env['HTTP_ACCEPT_LANGUAGE'].scan(/^[a-z]{2}/).first
+    end
```

Comment:
恐らく [Rails ガイド](https://guides.rubyonrails.org/i18n.html#choosing-an-implied-locale) の内容を参考にされたと思うので間違いとかではないのですが、 Ruby では引数の無いメソッドを変数に見せかける事が多いです。
思い切って以下のように書いても良いかもしれないですね。

```suggestion
  def switch_locale(&action)
    I18n.with_locale(current_locale, &action)
  end

  private

    def current_locale
      request.env['HTTP_ACCEPT_LANGUAGE'].scan(/^[a-z]{2}/).first
    end
```

Reply:
サンプルコードをそのまま記載していました。
そのまま利用させていただきます。
本来ならば、提供可能なi18nから選択できることを検討すべきですね。
本PRに関しては主題がずれるので、いったんi18n対応を除外させていただきます。